### DISCUSSION TITLE
More dynamic default content types

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [OrchardCMS/OrchardCore#11259] [Q&A]
Hello,

Started in the _gitter.im_, looking for a way for the default _agency_ content types to be more dynamic than the static items they are out of the box, if at all possible.

I have not explored yet, but perhaps this is a function of `Creatable`, `Listable` and so forth?

I'm not sure that is readily evident through the description in terms of what to expect, i.e. _Determines if an instance of this content type can be `[created|listed]` through the UI._

**Comment by devlife** [comment]
Because those content types (`Service`, `Project`, etc) are _not_ listed as `Creatable` or `Listable` because the `Landing Page` uses the `BagPart` instead of a `Content Item Picker`. With the `BagPart` you create the `ContentItem`s inline in the container by clicking the blue `Add Item` button. In this case, the page. If it used a `Content Item Picker` then you would create the `ContentItem` via the `Content Items` -> `New` button and then _select_ those `ContentItem`s from the page.

![image](https://user-images.githubusercontent.com/237369/155636679-ba6ff407-ca86-4266-b889-d6c5cc5f3bfa.png)
![image](https://user-images.githubusercontent.com/237369/155636698-ee71e805-6307-4ce3-91b2-7718e81d0a2f.png)
![image](https://user-images.githubusercontent.com/237369/155636723-f68276f9-d5a6-4991-a388-e5523614b248.png)

**Reply by mwpowellhtx** [reply]
Okay, so this is the reason they are what they are out of the box. So if we wanted to support a more dynamic use case than static bagged content, we would need a sort of `Content Item Picker`? Then of course enable some of those options, `Creatable`, `Listable`, etc. Might be closer to what might be more productive, I think. But to be truly dynamic, I think we might also want to support formatted input, perhaps JSON or XML, let's say, feeding the maintenance side of things.

**Comment by mwpowellhtx** [comment]
Also sort of tangential to that, how do we know what _css_ is involved in the picture? Is this documented somewhere? i.e. `fa-shopping-cart`, `fa-laptop`, `fa-lock`, and so on. Not finding the _css_ references anywhere in the admin configuration, per se. We can define our own _css_ styles? Perhaps I overlooked it.

**Reply by mwpowellhtx** [reply]
For example the [`.io` URL](https://docs.orchardcore.net/en/dev/docs/reference/modules/AdminMenu/#font-awesome-icon-picker) appears to be broken... don't know how to connect the dots between the _style_ and the _resource_, _iconage_, etc, if you will.

**Reply by mwpowellhtx** [reply]
The comment is accurate, _The icon css class from font-awesome. e.g., fa-laptop, fa-shopping-cart, fa-lock_, absent supporting docs, however. Also the source repo appears to have been archived, not sure if there is a migration path or alternative readily available. Open to suggestions.

**Reply by mwpowellhtx** [reply]
Changing from _standard_ to _icon picker_ help a bit. The concern over the state of the source repo, however, still a concern.

**Reply by Skrypt** [reply]
Can you send us a screenshot of what is not working? Preferably a gif or mpg so that we can see what's broken. I've tested the Icon Picker locally as the editor option on a text field and it seems to work fine from here. Of course, if it's not displaying properly on the frontend side that means you need to have FontAwesome installed on your frontend theme too.

**Reply by mwpowellhtx** [reply]
I explained what is _"not working"_, the caveats, etc. It 'works', better if you use the _picker_, at least you have an idea what you are getting, short of a _css_ tag cross reference. However, the source appears to be obsolete, archived, last update perhaps 2-3 yrs ago at the latest.

**Reply by Skrypt** [reply]
I will try with The Agency theme. Maybe it's an issue that is only related with the FlowPart.

**Reply by Skrypt** [reply]
Ok, I get it now. The FontAwesome IconPicker javascript library has been orphaned by its creator.

https://github.com/itsjavi/fontawesome-iconpicker

But, it's been archived there and the documentation doesn't point to a proper URL.

https://itsjavi.com/fontawesome-iconpicker/

**Reply by Skrypt** [reply]
I think at some point we changed the documentation to point to a fork and then the fork doesn't exist anymore.
It's not a big deal if the Icon Picker still works ... at least for now.

**Reply by Skrypt** [reply]
There seem to be a new one with an active contributor here : https://furcan.github.io/IconPicker/
But, I personally prefer the old one. More options to work with.
